server/src/instant/util/storage.clj

```diff
@@ -0,0 +1,207 @@
+(ns instant.util.storage
+  (:require [clojure.string :as string]
+            [instant.util.exception :as ex]
+            [instant.model.app-user :as app-user-model]
+            [instant.model.rule :as rule-model]
+            [instant.model.app-user-refresh-token :as app-user-refresh-token-model]
+            [instant.db.cel :as cel]
+            [instant.util.json :refer [->json <-json]]
+            [instant.storage.s3 :as s3-util]
+            [instant.storage.beta :as storage-beta])
+
+  (:import
+   (java.util UUID)))
+
+
+;; scopes filename to app-id directory
+(defn ->object-key [app-id filename]
+  (str app-id "/" filename))
```

Comment:
I wonder if we'll run into problems with rate limits for partitioned prefixes: https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html

We could probably ask the user to add their own prefixes if we run into that problem, though.

Reply:
nice catch, good thing to keep an eye out for!